tmt/steps/execute/internal.py

```diff
@@ -10,25 +9,46 @@
 import tmt
 import tmt.utils
 from tmt.steps.execute import TEST_OUTPUT_FILENAME
+from tmt.steps.provision import DEFAULT_RSYNC_OPTIONS
 from tmt.steps.provision.local import GuestLocal
 
-REBOOT_VARIABLES = [
+REBOOT_VARIABLES = (
     "TMT_REBOOT_COUNT",
     "REBOOTCOUNT",
-    "RSTRNT_REBOOTCOUNT",
-    ]
-REBOOT_TYPE = "reboot"
+    "RSTRNT_REBOOTCOUNT")
 REBOOT_SCRIPT_PATHS = (
     "/usr/local/bin/rstrnt-reboot",
     "/usr/local/bin/rhts-reboot",
     "/usr/local/bin/tmt-reboot")
-REBOOT_BACKUP_EXT = ".backup"
+REBOOT_REQUEST_FILENAME = "reboot_request"
 REBOOT_SCRIPT = f"""\
 #!/bin/sh
-echo "{{token}}:{{{{\
-    \\"type\\": \\"{REBOOT_TYPE}\\",\
-    \\"version\\": \\"0.2\\"\
-}}}}"
+touch "$TMT_TEST_DATA/{REBOOT_REQUEST_FILENAME}"
+"""
+REBOOT_BACKUP_EXT = ".backup"
+REBOOT_SETUP_NAME = "reboot_setup"
+REBOOT_SETUP_SCRIPT = f"""\
+#!/bin/sh
+for file in {" ".join(REBOOT_SCRIPT_PATHS)}; do
+    # Backup existing file, keep track of what has been backed-up
+    if [ -e "$file" ]; then
+        mv "$file" "$file{REBOOT_BACKUP_EXT}"
+        echo "$file"
+    fi
+    cp "$1" "$file" && chmod +x "$file"
+done
+"""
+REBOOT_TEARDOWN_NAME = "reboot_teardown"
+REBOOT_TEARDOWN_SCRIPT = f"""\
+#!/bin/sh
+for file in {" ".join(REBOOT_SCRIPT_PATHS)}; do
+    rm "$file"
+done
+
+# We pass back the backed up files
```

Comment:
Minor thought (feel free to ignore): Is this needed? What about just checking if the backup files exist during the loop above and forget about storing the backed up files?

Reply:
Thought about it as well but I feel like this is more robust - we only clean our own mess :-)